Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core.UnitTests/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRuleTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.Contracts.Core.UnitTests.AttachAnnexToBindingContract.BusinessRules;
+
+using Core.AttachAnnexToBindingContract.BusinessRules;
+using EvolutionaryArchitecture.Fitnet.Common.Core.BusinessRules;
+
+public sealed class AnnexCanOnlyBeAttachedToActiveBindingContractRuleTests
+{
+    [Fact]
+    internal void Given_attach_annex_When_binding_contract_terminated_Then_it_is_not_possible_to_attach()
+    {
+        // Arrange
+        var now = DateTimeOffset.UtcNow;
+        var bindingContractTerminatedAt = now.AddDays(-1);
+        var bindingContractExpiringAt = now.AddYears(1);
+
+        // Act && Assert
+        ShouldThrowException(() =>
+            BusinessRuleValidator.Validate(
+                new AnnexCanOnlyBeAttachedToActiveBindingContractRule(
+                    bindingContractTerminatedAt,
+                    bindingContractExpiringAt,
+                    now)));
+    }
+
+    [Fact]
+    internal void Given_attach_annex_When_binding_contract_expired_Then_it_is_not_possible_to_attach()
+    {
+        // Arrange
+        var now = DateTimeOffset.UtcNow;
```

Comment:
maybe we can set as private readonly field. To not duplicate in each test

Reply:
Done